File: app/src/main/java/com/lessonscontrol/data/viewModel/LessonViewModel.java
package com.lessonscontrol.data.viewModel;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;

import com.lessonscontrol.data.entities.Lesson;
import com.lessonscontrol.data.entities.Student;
import com.lessonscontrol.data.repositories.LessonRepository;

import java.util.List;

/**
 * Student View Model.
 *
 * @author hblonski
 */
public class LessonViewModel extends AndroidViewModel {

    private LessonRepository lessonRepository;

    public LessonViewModel(Application application) {
        super(application);
        lessonRepository = new LessonRepository(application);
    }

    public void insert(Lesson lesson) {
        lessonRepository.insert(lesson);
    }

    public void update(Lesson lesson) {
        lessonRepository.update(lesson);
    }

    public void delete(Lesson lesson) {
        lessonRepository.delete(lesson);
    }

    public LiveData<List<Lesson>> findLessonsByStudent(Student student) {
        return lessonRepository.findLessonsByStudent(student);
    }
}

File: app/src/main/java/com/lessonscontrol/data/viewModel/LessonViewModel.kt
package com.lessonscontrol.data.viewModel

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData

import com.lessonscontrol.data.entities.Lesson
import com.lessonscontrol.data.entities.Student
import com.lessonscontrol.data.repositories.LessonRepository

/**
 * Student View Model.
 *
 * @author hblonski
 */
class LessonViewModel(application: Application) : AndroidViewModel(application) {

    private val lessonRepository: LessonRepository = LessonRepository(application)

    fun insert(lesson: Lesson) {
        lessonRepository.insert(lesson)
    }

    fun update(lesson: Lesson) {
        lessonRepository.update(lesson)
    }

    fun delete(lesson: Lesson) {
        lessonRepository.delete(lesson)
    }

    fun findLessonsByStudent(student: Student): LiveData<List<Lesson>> {
        return lessonRepository.findLessonsByStudent(student)
    }
}
